usr/sbin/pkcscca/pkcscca.c

```diff
@@ -930,14 +929,51 @@ int replace_keys(CK_FUNCTION_LIST *funcs, CK_SESSION_HANDLE sess,
     return 0;
 }
 
-int cca_migrate_asymmetric(struct key *key, char **out, struct algo algo)
+int cca_migrate_asymmetric(struct key *key, char **out, struct algo algo,
+                           int masterkey)
 {
     long return_code, reason_code, exit_data_length, key_identifier_length;
     unsigned char *key_identifier;
 
     exit_data_length = 0;
     key_identifier_length = key->attr_len;
 
+    if (key->attr_len < 16 ||
+        key->opaque_attr[0] == 0x1e) { /* 0x1e: external PKA token */
+        printf("Skipping key, its a public key. label=%s, handle=%lu\n",
```

Comment:
Just while seeing this: maybe it makes sense to introduce #define's for these CCA specific section indications: 0x1e = external key token, 0x20 = private section, 0x21 = public section etc. But we saw that in opencryptoki the hex values are mostly used directly. Perhaps something for a housekeeping commit later.

Reply:
Yes, if we do that it should be common defines used for the CCA token and this tool. 
Not in this PR.....